streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableMapValues.java

```diff
@@ -81,6 +81,18 @@ private V1 computeValue(final K key, final V value) {
         return newValue;
     }
 
+    private ValueAndTimestamp<V1> computeValueAndTimestamp(final K key, final ValueAndTimestamp<V> valueAndTimestamp) {
+        V1 newValue = null;
+        long timestamp = 0;
+
+        if (valueAndTimestamp != null) {
+            newValue = mapper.apply(key, valueAndTimestamp.value());
+            timestamp = valueAndTimestamp.timestamp();
+        }
+
+        return ValueAndTimestamp.make(newValue, timestamp);
```

Comment:
This will always produce a non-null ValueAndTimestamp result, possibly with a `null` value and a `0` timestamp, if it wasn't resolved upstream... Is this right?

Reply:
No. `ValueAndTimestamp` will never carry a `null` value. `make` would return `null` if `newValue` is null.

Reply:
Ah, my mistake!